app/src/org/commcare/fragments/personalId/PersonalIdPhoneFragment.java

```diff
@@ -148,11 +156,52 @@ private void displayPhoneNumber(String fullNumber) {
     }
 
     private void onContinueClicked() {
-        // TODO: Trigger API call for phone number validation and registration api later
-        navigateToBiometricSetup();
+        String phone = PhoneNumberHelper.buildPhoneNumber(
+                binding.countryCode.getText().toString(),
+                binding.connectPrimaryPhoneInput.getText().toString()
+        );
+
+        new PersonalIdApiHandler() {
+            @Override
+            protected void onSuccess(PersonalIdSessionData sessionData) {
+                viewModel.setPersonalIdSessionData(sessionData);
+                if (viewModel.getPersonalIdSessionData().getToken() != null) {
+                    onConfigurationSucesss();
+                } else { // This is called when api returns success but with a a failure code
+                    onConfigurationFailure();
+                }
+            }
+
+            @Override
+            protected void onFailure() {
+                navigateFailure();
+            }
+        }.makeStartConfigurationCall(requireActivity(), phone);
+    }
+
+
+    private void onConfigurationSucesss() {
+        Navigation.findNavController(binding.personalidPhoneContinueButton).navigate(navigateToBiometricSetup());
+    }
+
+    private void onConfigurationFailure() {
+        Logger.log(LogTypes.TYPE_USER, viewModel.getPersonalIdSessionData().getSessionFailureCode());
+        Navigation.findNavController(binding.personalidPhoneContinueButton).navigate(navigateToMessageDisplay());
```

Comment:
think this message should be non-cancellable or we should just finish app and exit if user click ok on the message screen.